airflow/providers/google/cloud/transfers/gcs_to_bigquery.py

```diff
@@ -544,43 +494,272 @@ def _check_schema_fields(self, table_resource):
         :param table_resource: Configuration or table_resource dictionary
         :return: table_resource: Updated table_resource dict with schema_fields
         """
-        if not self.autodetect and not self.schema_fields:
-            raise RuntimeError(
-                "Table schema was not found. Set autodetect=True to "
-                "automatically set schema fields from source objects or pass "
-                "schema_fields explicitly"
-            )
-        elif not self.schema_fields:
+        if not self.schema_fields:
```

Comment:
This is still broken in multiple places. 

For example, it makes the assumption about the delimiter being "," instead of the value of `field_delimiter` parameter, same for `quote_character` parameter.

Overall, I don't think that attempt to replicate the logic of determining schema instead of just letting "autodetect" and BigQuery do their thing is a wise idea - there are numerous edge cases that we would need to cover and we will continue getting issues because of implementations getting out of sync.

I believe that the good approach would be to remove the explicit setting of `"schema"` in configuration if `autodetect==True` or remove `_check_schema_fields()` altogether.

Reply:
Yes, seems like that. I was trying to add this logic to simplify working with BigQuery schemas: as you know, BQ can only determine schema fields if the values for every column is of non-string format. (please, check the link: https://cloud.google.com/bigquery/docs/schema-detect#csv_header)
But turned out that my implementation needs a lot of reworking :)
I will remove this logic and let the schema fields be "string_field_1" as it was before.